pipeline/extract.py

```diff
@@ -123,21 +123,41 @@ def parse_pdf(court_case: dict):
     first_page = reader.pages[0].extract_text()
     second_page = reader.pages[1].extract_text()
     last_page = reader.pages[-1].extract_text()
-    judge = re.search(r"(?<=Before :\n)([A-Z].*)", first_page)
-    if not judge:
-        judge = re.search(r"(?<=Before  : \n \n)([A-Z].*)", first_page)
-    judge = judge.group(1)
-    court_case["judge_name"] = judge
-
-    court_case["case_no"] = re.search(
-        r"(?<=Case No: )([A-Z, -].*)", first_page).group(1)
-
-    court_case["date"] = re.search(
-        r"(?<=Date: )([0-9].*)", first_page).group(1)
-
-    court_case["introduction"] = second_page
-
-    court_case["conclusion"] = last_page
+    try:
+        judge = re.search(r"(?<=Before :\n)(.*)", first_page)
+        if not judge:
+            judge = re.search(r"(?<=Before  : \n \n)(.*)", first_page)
+        if not judge:
+            judge = re.search(r"(?<=Before : \n \n)(.*)", first_page)
+        if not judge:
+            judge = re.search(r"(?<=Before:\n)(.*)", first_page)
+        if not judge:
+            judge = re.search(r"(?<=BEFORE:\n)(.*)", first_page)
+        if not judge:
+            judge = re.search(r"(THE HONOURABLE.*)", first_page)
+
```

Comment:
Could you not put all your regex expressions together and just separate with a `|` ?

Reply:
I'll look into it a bit more! I tried it at first and I kept getting an error where the 'positive lookahead' part of my regex was of variable length. However, I'll keeping changing things around to see if i can work it out!